### DISCUSSION TITLE
ComboboxItem causes extra rerenders

### DISCUSSION THREAD

**Discussion by pownkel** [ariakit/ariakit#4114] [Questions]
I'm working off of the [Textarea with inline combobox](https://ariakit.org/examples/combobox-textarea) example, and I'm seeing some strange behavior. In the example code, I changed the `render` prop to a function that contains a `console.log` line, so I could see how often the base component is rerendered. When the combobox opens, if there are any `ComboboxItems` onscreen, `render()` is called 6 times. If I remove the `ComboboxItem` components, `render()` is called only 4 times. The number of rerenders is the same no matter how many `ComboboxItems` there are, as long as there is at least one, and hardcoding the props to constant values has no effect. The extra rerenders seem purely dependent on whether or not a `ComboboxItem` is present

In a larger, more complex application, this is somehow causing dozens of additional rerenders just from having a single `ComboboxItem` onscreen.  Virtualizing the list doesn't help, and memoizing props has no effect, so I've been struggling to find a way to improve performance. What could be causing these rerenders? Is there a state update happening behind the scenes, and how much of the component tree should it affect?

**Comment by benrodrs** [comment]
Renders shouldn't be a problem. Could you please provide a StackBlitz example showing the performance issue?

**Reply by pownkel** [reply]
I can't share the app that has performance issues, my minimal repro is the example code with a few changes that I explained above. There are only a few extra rerenders in that case, but I'm hoping that if I can understand why those are happening, I can figure out why the rerenders are compounding so much for my larger use case

**Comment by pownkel** [comment]
Updating this question: I still haven't figured out why the combobox being open caused so many rerenders, but it turned out I had a bigger performance issue in the Formik form that the combobox was a part of, which was rerendering on every combobox update. Once I fixed that, I stopped seeing slowdowns on the page.
